### PR TITLE
Added C++ example OffscreenRendering

### PR DESCRIPTION
**Pull request by prewettg**
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/2914)

### CODE REVIEW

File: examples/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,9 @@ EXAMPLE_CPP(ISSKeypoints              ${CMAKE_PROJECT_NAME})
 EXAMPLE_CPP(LineSet                   ${CMAKE_PROJECT_NAME})
 EXAMPLE_CPP(Log                       ${CMAKE_PROJECT_NAME})
 EXAMPLE_CPP(OdometryRGBD              ${CMAKE_PROJECT_NAME})
-EXAMPLE_CPP(Octree                    ${CMAKE_PROJECT_NAME})
+if (BUILD_GUI)
```
Comment (germanros1987) [inline examples/cpp/CMakeLists.txt:47]: Why did the octree go out?

**Review by prewettg — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/2914)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @errissa and @germanros1987)

___
*[examples/cpp/CMakeLists.txt, line 47 at r1](https://reviewable.io/reviews/intel-isl/open3d/2914#-MRMxTzvJyu_XZ-rjGt_-r1-47:-MRSNLqT7yz0HXPkSzWD:bw20id4) ([raw file](https://github.com/intel-isl/open3d/blob/cc2168f525ab5aeb4ec6ed0bb3681a0ea835b2ca/examples/cpp/CMakeLists.txt#L47)):*
<details><summary><i>Previously, germanros1987 wrote…</i></summary><blockquote>

Why did the octree go out?
</blockquote></details>

Oops, fixed

**Review by errissa — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/2914)* status: all files reviewed, 1 unresolved discussion (waiting on @germanros1987)

File: examples/cpp/CMakeLists.txt (hunk 1 above)
Comment (lichangW) [inline examples/cpp/CMakeLists.txt:48]: BUILD_GUI should be OFF?

Comment (nubertj) [reply examples/cpp/CMakeLists.txt:48]: should it not be off, when testing the offscreen rendering?
Thanks a lot!

**Pull request merged**